manifests/globals.pp

```diff
@@ -95,72 +96,72 @@
+  Optional[String[1]] $server_package_name         = undef,
+  Optional[String[1]] $contrib_package_name        = undef,
+  Optional[String[1]] $devel_package_name          = undef,
+  Optional[String[1]] $java_package_name           = undef,
+  Optional[String[1]] $docs_package_name           = undef,
+  Optional[String[1]] $perl_package_name           = undef,
+  Optional[String[1]] $plperl_package_name         = undef,
+  Optional[String[1]] $plpython_package_name       = undef,
+  Optional[String[1]] $python_package_name         = undef,
+  Optional[String[1]] $postgis_package_name        = undef,
 
-  $service_name                                    = undef,
-  $service_provider                                = undef,
-  $service_status                                  = undef,
-  $default_database                                = undef,
+  Optional[String[1]] $service_name                = undef,
+  Optional[String[1]] $service_provider            = undef,
+  Optional[String[1]] $service_status              = undef,
+  Optional[String[1]] $default_database            = undef,
 
-  $validcon_script_path                            = undef,
+  Optional[String[1]] $validcon_script_path        = undef,
 
-  $initdb_path                                     = undef,
-  $createdb_path                                   = undef,
-  $psql_path                                       = undef,
-  $pg_hba_conf_path                                = undef,
-  $pg_ident_conf_path                              = undef,
-  $postgresql_conf_path                            = undef,
+  Optional[String[1]] $initdb_path                 = undef,
+  Optional[String[1]] $createdb_path               = undef,
+  Optional[String[1]] $psql_path                   = undef,
+  Optional[String[1]] $pg_hba_conf_path            = undef,
+  Optional[String[1]] $pg_ident_conf_path          = undef,
+  Optional[String[1]] $postgresql_conf_path        = undef,
   Optional[Stdlib::Filemode] $postgresql_conf_mode = undef,
-  $recovery_conf_path                              = undef,
-  $default_connect_settings                        = {},
+  Optional[String[1]] $recovery_conf_path          = undef,
+  Hash $default_connect_settings                   = {},
 
-  $pg_hba_conf_defaults                            = undef,
+  Optional[Boolean] $pg_hba_conf_defaults          = undef,
 
-  $datadir                                         = undef,
-  $confdir                                         = undef,
-  $bindir                                          = undef,
-  $xlogdir                                         = undef,
-  $logdir                                          = undef,
-  $log_line_prefix                                 = undef,
-  $manage_datadir                                  = undef,
-  $manage_logdir                                   = undef,
-  $manage_xlogdir                                  = undef,
+  Optional[String[1]] $datadir                     = undef,
```

Comment:
It should be possible to use Stdlib::Absolutepath here. Is also an option for all the other parameters which can contain a path

Reply:
Done